src/herder/TxSetUtils.cpp

```diff
@@ -144,15 +144,6 @@ TxSetUtils::getInvalidTxList(TxSetTransactions const& txs, Application& app,
     ZoneScoped;
     LedgerTxn ltx(app.getLedgerTxnRoot(), /* shouldUpdateLastModified */ true,
                   TransactionMode::READ_ONLY_WITHOUT_SQL_TXN);
-#ifdef BUILD_TESTS
-    if (protocolVersionIsBefore(ltx.loadHeader().current().ledgerVersion,
-                                ProtocolVersion::V_20))
-    {
-        return {};
-    }
-#endif
-    releaseAssert(protocolVersionStartsFrom(
```

Comment:
I'm ok with removing this, but can we make sure this can actually happen? Specifically, it is a bit surprising to me that a node tries to nominate a tx set that is targeting the old protocol. Do out-of-sync nodes think that they're in-sync and thus should try to nominate tx sets?

Reply:
Yea, it depends on how a node starts. If a validator is started without `--wait-for-conseus`, it'll consider itself "synced", and attempt nominating a value for LCL+1 to potentially unstick the network. For ledger 2, it doesn't matter of course, so we just timeout and start buffering ledgers from the network.

I should add that right now the code is written such that we won't trigger getInvalidTxs crash (as it's not called on empty tx sets), but I think this is still a valid scenario to consider.